relay-metrics/src/protocol.rs

```diff
@@ -516,10 +510,7 @@ impl Metric {
 
         let name_value_str = components.next()?;
         let ty = components.next().and_then(|s| s.parse().ok())?;
-        let (name_and_namespace, unit, value) = parse_name_unit_value(name_value_str, ty)?;
-        let (raw_namespace, name) = name_and_namespace
-            .split_once('/')
-            .unwrap_or(("custom", string));
+        let (raw_namespace, name, unit, value) = parse_name_unit_value(name_value_str, ty)?;
```

Comment:
I had to refactor these parser functions because `is_valid_metric_name` no longer allows `/` (because it is also called from normalization). Instead of validate-then-split, we know do split-then-validate.